tests/Jobs/ImportRockTheVoteReportTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Tests\Http\Web;
+
+use Tests\TestCase;
+use Chompy\User;
+use Chompy\Jobs\ImportFileRecords;
+use Illuminate\Support\Facades\Bus;
+use Chompy\Models\RockTheVoteReport;
+use Illuminate\Support\Facades\Storage;
+use Chompy\Jobs\ImportRockTheVoteReport;
+
+class ImportRockTheVoteReportTest extends TestCase
+{
+    /**
+     * Test that report is not downloaded when status not complete.
+     *
+     * @return void
+     */
+    public function testReportStatusNotComplete()
+    {
+        Bus::fake();
+
+        $report = factory(RockTheVoteReport::class)->create();
+        $user = User::forceCreate(['role' => 'admin']);
+
+        $this->rockTheVoteMock->shouldReceive('getReportStatusById')->andReturn((object) [
+            'status'=> 'building',
+            'record_count' => 117,
+            'current_index' => 3,
+        ]);
+        $this->rockTheVoteMock->shouldNotReceive('getReportByUrl');
+
+        $importRockTheVoteReportJob = new ImportRockTheVoteReport($user, $report);
+
+        $importRockTheVoteReportJob->handle();
+
+        $this->assertEquals($report->status, 'building');
+        $this->assertEquals($report->row_count, 117);
+        $this->assertEquals($report->current_index, 3);
+        $this->assertEquals($report->user_id, $user->northstar_id);
+
+        Bus::assertDispatched(ImportRockTheVoteReport::class, function ($job) use (&$report, &$user) {
+            $params = $job->getParameters();
+
+            $this->assertEquals($report, $params['report']);
+            $this->assertEquals($user, $params['user']);
+
+            return true;
```

Comment:
This might make more sense as the "success criteria" for the [bus assertion](https://laravel.com/docs/5.7/mocking#bus-fake):

```php
Bus::assertDispatched(ImportRockTheVoteReport::class, function ($job) use (&$report, &$user) {
    $params = $job->getParameters();
    
    return $params['report'] == $report && $params['user'] == $user;
});

```